source/type/vehicle_journey.cpp

```diff
@@ -267,31 +278,49 @@ std::set<StopPoint*> VehicleJourney::get_sections_stop_points(const StopArea* st
      *  - s1/s2 from the 5th to the 10th,
      *  - s1/s2/s5 from the 11th to the 15th
      * */
-    std::set<StopPoint*> res;
+    std::set<uint16_t> res;
     boost::optional<uint16_t> start_idx;
+    boost::optional<uint16_t> last_area_idx;
+    bool section_ending = false;
     const auto* base_vj = this->get_corresponding_base();
     const auto* vj = base_vj ? base_vj : this;
     for (const auto& st : vj->stop_time_list) {
         // The line section is using stop_areas so we make sure we have one
         if (!st.stop_point || !st.stop_point->stop_area) {
             continue;
         }
+        const auto* sa = st.stop_point->stop_area;
+
+        // Must close section before potentially starting a new one
+        if (section_ending && *last_area_idx != sa->idx) {
```

Comment:
At that point `last_area_idx` is always `end_stop->idx`so IMHO this looks easier to understand:
```suggestion
        if (section_ending && *end_stop->idx != sa->idx) {
```

Reply:
I simplified it, let me know if it's ok for you